src/components/QueryEditor/AnnotationQueryEditor.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+
+import { css } from 'emotion';
+import { AnnotationQuery } from '@grafana/data';
+import { Input, QueryField } from '@grafana/ui';
+
+import { OpenSearchQuery } from '../../types';
+import { OpenSearchQueryEditorProps } from '.';
+
+type Props = OpenSearchQueryEditorProps & {
+  annotation?: AnnotationQuery<OpenSearchQuery>;
+  onAnnotationChange?: (annotation: AnnotationQuery<OpenSearchQuery>) => void;
+};
+
+const styles = {
+  marginRight: css`
+    margin-right: 4px;
+  `,
+  queryWrapper: css`
+    display: flex;
+    flex-grow: 1;
+  `,
+};
+
+export function OpenSearchAnnotationsQueryEditor(props: Props) {
+  const annotation = props.annotation!;
+  const onAnnotationChange = props.onAnnotationChange!;
+
+  return (
+    <>
+      <div className="gf-form-group">
+        <div className={styles.queryWrapper}>
+          <QueryField
+            query={annotation.target.query}
```

Comment:
```suggestion
            query={annotation.target?.query}
```
or we might get something like this.. :D 

![Screenshot 2022-06-09 at 17 49 21](https://user-images.githubusercontent.com/90795735/173039640-2957b309-03eb-4a70-a32f-947f502573b5.png)